tests/test_arrow_dataset.py

```diff
@@ -830,6 +926,23 @@ def test_map_multiprocessing(self, in_memory):
                     self.assertNotEqual(dset_test._fingerprint, fingerprint)
                     assert_arrow_metadata_are_synced_with_dataset_features(dset_test)
 
+        with tempfile.TemporaryDirectory() as tmp_dir:  # num_proc > len(dset)
+            with self._create_dummy_dataset(in_memory, tmp_dir) as dset:
+                self.assertDictEqual(dset.features, Features({"filename": Value("string")}))
+                fingerprint = dset._fingerprint
+                dset = dset.select([0, 1])
```

Comment:
Here to avoid writing a new file for the indices that we would have to close for the windows CI, you can just keep the indices in memory:
```suggestion
                dset = dset.select([0, 1], keep_in_memory=True)
```

Hopefully it fixes the windows CI